### PR TITLE
Issue #528 follow-up: fix separate-emission summaries

### PR DESCRIPTION
**Pull request by stranske**
> **Source:** Issue #528

#### Scope
Lodging and transport are the most operationally consequential inventory categories for early planning. Once source adapters and normalized option contracts exist, the repo needs a first ingestion scaffold that can turn raw source snapshots into normalized lodging and transport options with provenance, feasibility, and source-confidence information attached.

Parent epic: #525

### Related Issues/PRs
- [#525](https://github.com/stranske/trip-planner/issues/525)
- [#521](https://github.com/stranske/trip-planner/issues/521)
- [#522](https://github.com/stranske/trip-planner/issues/522)
- [#526](https://github.com/stranske/trip-planner/issues/526)
- [#527](https://github.com/stranske/trip-planner/issues/527)

#### Tasks
- [x] Implement pipeline scaffolding that accepts raw lodging and transport snapshots, applies entity resolution or deduplication, and emits normalized `LodgingOption` and `TransportOption` objects.
- [x] Ensure the pipeline records source provenance, freshness, unresolved conflicts, and normalization warnings alongside the normalized outputs.
- [x] Add representative ingestion fixtures for at least one lodging source and one transport source, plus one case with duplicate or conflicting records.
- [x] Add tests covering happy-path normalization, duplicate handling, and degraded-source cases.
- [x] Document the pipeline stages so later live adapters can plug in without reworking the normalized object layer.
- [x] Expose enough pipeline summary metadata that later candidate-generation logic can filter out unusable or low-confidence outputs.

#### Acceptance criteria
- [x] The repo contains working ingestion scaffolding for lodging and transport snapshots into normalized option objects.
- [x] Provenance, normalization warnings, and unresolved conflicts remain visible in the emitted results.
- [x] Tests cover at least one clean path and one degraded or duplicate path for both lodging and transport.
- [x] Pipeline stages are documented clearly enough for later adapter implementation work.
- [x] The output shape is compatible with issues #521 and #522 rather than inventing alternative option objects.

#### Follow-up disposition
Verifier concerns on merged PR #614 were resolved in this bounded follow-up:
- `keep_separate` / `needs_review` lodging and transport outputs now get stable unique option ids instead of colliding ids
- merged-away records no longer inflate `filtered_record_ids`
- `skipped_records` now reflects truly filtered-out records only

#### Validation
- `pytest tests/ingestion/test_lodging_pipeline.py tests/ingestion/test_transport_pipeline.py`
- `ruff format`

**Head SHA:** a9e317e3e9039353d15d9fb001be8ec73b154c00
**Merge commit:** 125af5189f2fbf48ce55e0c397ae6e50f01baf5f

### CODE REVIEW

File: trip_planner/ingestion/transport_pipeline.py [hunk 1]
```diff
@@ -173,7 +211,7 @@ def ingest_transport_snapshot(
     summary = IngestionSummary(
         total_records=len(snapshot.records),
         emitted_options=len(transport_options),
-        skipped_records=max(0, len(snapshot.records) - len(transport_options)),
+        skipped_records=len(filtered_record_ids),
         degraded_options=sum(1 for option in transport_options if option.notes),
```
[the remark this answers is not in the document]
Comment (stranske) [reply trip_planner/ingestion/transport_pipeline.py]: Aligned with the PR intent in `06e9b8c3`: merged records are no longer treated as skipped, and the updated lodging/transport tests now assert `skipped_records == 0` for merge decisions.

**Comment by stranske** [comment]
Addressed the Copilot review points on `a9e317e3`.

- merge decisions no longer contribute merged-away record IDs to `filtered_record_ids`
- `skipped_records` now stays aligned with truly suppressed records only
- updated lodging/transport tests to assert the merge-path summary behavior
- ran `pytest tests/ingestion/test_lodging_pipeline.py tests/ingestion/test_transport_pipeline.py`
- applied `ruff format` after CI flagged formatting

**Comment by stranske** [comment]
[issue-completion-audit][ica-component1-gap] Component 1 gap: Automated Status Summary checklist still shows incomplete task/acceptance items. Please sync PR summary evidence to completed work.
